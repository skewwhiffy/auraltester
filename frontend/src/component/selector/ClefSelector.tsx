import H1 from "../H1";
import Centre from "../Centre";
import FancyRadioButtons from "../FancyRadioButtons";

export default ClefSelector

interface Props {
  title?: string | null
  value?: string | null

  onChange(value: string): void
}

function ClefSelector({title, value, onChange}: Props) {
  return (
    <>
      {title && <Centre><H1>{title}</H1></Centre>}
      <Centre>
        <FancyRadioButtons value={value} onChange={onChange}>
          <FancyRadioButtons.Item>Bass</FancyRadioButtons.Item>
        </FancyRadioButtons>
      </Centre>
    </>
  )
}
