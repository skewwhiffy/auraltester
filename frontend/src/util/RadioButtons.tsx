import React from 'react'
import { Form } from "react-bootstrap";

type OnChangeHandler = (value: string) => void

export interface RadioButtonDefinition {
  key?: string,
  value: string,
  label?: string
}

export interface Props {
  values: Array<RadioButtonDefinition>,
  name: string,
  defaultValue?: string,
  onChange: OnChangeHandler
}

interface State {
  value: string
}

class RadioButtons extends React.Component<Props, State> {
  constructor(props: Props) {
    if (props.values.length === 0) {
      throw Error('Need at least one radio button')
    }
    super(props);
  }

  private onChange = (ev: any) => {
    (this.props.onChange || (_ => { }))(ev.target.value)
  }

  private renderRadioButton = (radioButton: RadioButtonDefinition) => {
    const value = radioButton.value
    const key = radioButton.key || value
    const label = radioButton.label || value
    const defaultValue = this.props.defaultValue === undefined
      ? this.props.values[0].value
      : this.props.defaultValue
    return (
      <Form.Check
        id={`${this.props.name}-${key}`}
        key={key}
        name={this.props.name}
        value={value}
        label={label}
        type='radio'
        defaultChecked={value === defaultValue}
      />
    )
  }

  render() {
    return (
      <Form.Group onChange={this.onChange}>
        {this.props.values.map(this.renderRadioButton)}
      </Form.Group>
    )
  }
}

export default RadioButtons