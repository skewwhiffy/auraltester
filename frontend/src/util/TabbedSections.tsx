import { Container, Nav } from "react-bootstrap"
import { Route, Routes } from "react-router"
import { Link } from "react-router-dom"
import { capitalizeFirstCharacter } from "."

export type TabbedSection = {
  getElement: () => JSX.Element
} & ({ path: string } | { text: string })

export interface Props {
  sections: TabbedSection[]
}

const getNavLink = (it: TabbedSection): JSX.Element => {
  const to = "path" in it ? it.path : "."
  const text = "text" in it ? it.text : capitalizeFirstCharacter(it.path)
  return (
    <Nav.Item key={to}>
      <Nav.Link as={Link} to={to}>{text}</Nav.Link>
    </Nav.Item>
  )
}

const getRoute = (it: TabbedSection): JSX.Element => {
  return "path" in it
    ? <Route key={it.path} path={it.path} element={it.getElement()} />
    : <Route key="." index element={it.getElement()} />
}

const TabbedSections = (props: Props): JSX.Element => {
  return (
    <Container>
      <Nav variant="tabs">
        {props.sections.map(getNavLink)}
      </Nav>
      <Routes>
        {props.sections.map(getRoute)}
      </Routes>
    </Container >
  )
}

export default TabbedSections