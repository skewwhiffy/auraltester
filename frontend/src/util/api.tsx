import axios from "axios"

interface ClefResponse {
  abc: string
}

const api = {
  async getClef(clef: string): Promise<ClefResponse> {
    const response = await axios.get("/api/clef", {
      params: {
        clef
      }
    })
    return response.data
  }
}

export default api