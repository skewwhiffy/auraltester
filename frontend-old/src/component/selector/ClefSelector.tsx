import { capitalizeFirstCharacter } from '../../util/helpers'
import RadioButtons, { RadioButtonDefinition } from '../../util/RadioButtons'
import { Col, Container, Row } from 'react-bootstrap'

type OnChangeHandler = (value: string) => void

export interface Props {
  value?: string
  onChange: OnChangeHandler
}

const ClefSelector = (props: Props): JSX.Element => {
  const values: RadioButtonDefinition[] = ['treble', 'alto', 'tenor', 'bass']
    .map(it => ({
      value: it,
      label: `${capitalizeFirstCharacter(it)} Clef`
    }))
  return (
    <Container>
      <Row><Col>Clef</Col></Row>
      <Row><Col>
        <RadioButtons
          values={values}
          name='clef'
          value={props.value}
          onChange={props.onChange}
        />
      </Col></Row>
    </Container>
  )
}

export default ClefSelector
