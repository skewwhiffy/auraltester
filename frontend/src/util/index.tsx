import api from './api'

const capitalizeFirstCharacter = (source: string): string => source.length === 0
  ? ''
  : `${source[0].toUpperCase()}${source.substring(1)}`

export { capitalizeFirstCharacter, api }
