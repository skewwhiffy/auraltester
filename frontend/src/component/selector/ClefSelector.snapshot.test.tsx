import ClefSelector from './ClefSelector'
import { Props } from '../../util/RadioButtons'
import renderer from 'react-test-renderer'

jest.mock('../../util/RadioButtons', () => (props: Props) => (
  <div>
    {`'${props.defaultValue}' '${props.name}' '${props.onChange}' '${JSON.stringify(props.values)}'`}
  </div>
))

describe('ClefSelector snapshot', () => {
  it('matches snapshot when default value set', () => {
    const actual = renderer.create(
      <ClefSelector defaultValue='default_value' onChange={() => { }} />
    )

    expect(actual).toMatchSnapshot()
  })

  it('uses treble as default value if not provided', () => {
    const actual = renderer.create(
      <ClefSelector onChange={() => { }} />
    )

    expect(actual).toMatchSnapshot()
  })
})