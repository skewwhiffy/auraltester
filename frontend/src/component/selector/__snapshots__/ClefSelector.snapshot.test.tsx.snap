// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClefSelector snapshot matches snapshot when default value set 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Clef
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        'default_value' 'clef' '() =&gt; {}' '[{"value":"treble","label":"Treble Clef"},{"value":"alto","label":"Alto Clef"},{"value":"tenor","label":"Tenor Clef"},{"value":"bass","label":"Bass Clef"}]'
      </div>
    </div>
  </div>
</div>
`;

exports[`ClefSelector snapshot uses treble as default value if not provided 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Clef
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        'treble' 'clef' '() =&gt; {}' '[{"value":"treble","label":"Treble Clef"},{"value":"alto","label":"Alto Clef"},{"value":"tenor","label":"Tenor Clef"},{"value":"bass","label":"Bass Clef"}]'
      </div>
    </div>
  </div>
</div>
`;
